Detectors/EMCAL/base/src/RCUTrailer.cxx

```diff
@@ -112,22 +114,52 @@ double RCUTrailer::getTimeSample() const
       tSample = 8.;
       break;
     default:
-      throw Error(Error::ErrorType_t::SAMPLINGFREQ_INVALID, (boost::format("Invalid sampling frequency value %d !") % int(fq)).str().data());
+      throw Error(Error::ErrorType_t::SAMPLINGFREQ_INVALID, fmt::format("Invalid sampling frequency value %d !", int(fq)).data());
   }
 
   return tSample * o2::constants::lhc::LHCBunchSpacingNS * 1.e-9;
 }
 
+void RCUTrailer::setTimeSample(double timesample) {
+  int fq = 0;
+  if(std::abs(timesample - 50) < DBL_EPSILON) fq = 0;
+  else if(std::abs(timesample - 100) < DBL_EPSILON) fq = 1;
+  else if(std::abs(timesample - 200) < DBL_EPSILON) fq = 2;
+  else throw Error(Error::ErrorType_t::SAMPLINGFREQ_INVALID, fmt::format("invalid time sample: %f", timesample).data());
+  mAltroCFG2 = (mAltroCFG2 & 0x1F) | fq << 5;
+}
+
 double RCUTrailer::getL1Phase() const
 {
   double tSample = getTimeSample(),
          phase = ((double)(mAltroCFG2 & 0x1F)) * o2::constants::lhc::LHCBunchSpacingNS * 1.e-9;
   if (phase >= tSample) {
-    throw Error(Error::ErrorType_t::L1PHASE_INVALID, (boost::format("Invalid L1 trigger phase (%e s (phase) >= %e s (sampling time)) !") % phase % tSample).str().data());
+    throw Error(Error::ErrorType_t::L1PHASE_INVALID, fmt::format("Invalid L1 trigger phase (%e s (phase) >= %e s (sampling time)) !", phase, tSample).data());
   }
   return phase;
 }
 
+void RCUTrailer::setL1Phase(double l1phase) {
+  int phase = l1phase/ 25.;
```

Comment:
If precision matters, better use https://github.com/AliceO2Group/AliceO2/blob/6b4aa8c47ba3918fcb14b66c62ba8821fab28d72/Common/Constants/include/CommonConstants/LHCConstants.h#L30

Reply:
No, the conversion in the hardware is done with 25 ns.